crypto/rand/drbg_lib.c

```diff
@@ -324,10 +329,46 @@ void RAND_DRBG_free(RAND_DRBG *drbg)
     CRYPTO_THREAD_lock_free(drbg->lock);
     CRYPTO_free_ex_data(CRYPTO_EX_INDEX_DRBG, drbg, &drbg->ex_data);
 
-    if (drbg->secure)
+    if (drbg->secure) {
+        OPENSSL_secure_clear_free(drbg->crngt_data, drbg->crngt_blocksize);
         OPENSSL_secure_clear_free(drbg, sizeof(*drbg));
-    else
+    } else {
+        OPENSSL_clear_free(drbg->crngt_data, drbg->crngt_blocksize);
         OPENSSL_clear_free(drbg, sizeof(*drbg));
+    }
+}
+
+/*
+ * Perform the continuous random number generator (CRNG) test.
+ * Each block of entropy is compared against the previous and the test
+ * fails if any pair is identical.  Any excess bytes are zeroed and not used.
+ * The DRBG's old entropy block is updated so the test can continue on the
+ * next call.  The block size is assumed to be non-zero, i.e. the test is
+ * executing.
```

Comment:
Consider adding "Returns 0 on failure, 1 on success" or similar.  Also, just to put it into a single comment, the function open curly needs to be on a line by itself. :)

Reply:
I only found one case of a misplaced curly (and one where I'd forgotten them).  Is there a second you noticed?

Reply:
Just the one for the function start.

Reply:
Thanks for the confirmation.